lib/chef/dsl/declare_resource.rb

```diff
@@ -119,7 +119,13 @@ def delete_resource(type, name, run_context: self.run_context)
       #
       def edit_resource!(type, name, created_at = nil, run_context: self.run_context, &resource_attrs_block)
         resource = find_resource!(type, name, run_context: run_context)
-        resource.instance_eval(&resource_attrs_block) if block_given?
+        if block_given?
```

Comment:
well you have to protect that because the block is optional and instance_eval/instance_exec will otherwise get confused...   not sure if that's what you were referring to though...